cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/StorageManager.java

```diff
@@ -0,0 +1,93 @@
+package com.linkedin.openhouse.cluster.storage;
+
+import static com.linkedin.openhouse.cluster.storage.StorageType.LOCAL;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * The StorageManager class is responsible for managing the storage types and providing the
+ * appropriate storage implementation based on the configuration.
+ */
+@Component
+public class StorageManager {
+
+  @Autowired StorageProperties storageProperties;
+
+  @Autowired StorageType storageType;
+
+  @Autowired List<Storage> storagesBeans;
+
+  /**
+   * Validate the storage properties.
+   *
+   * <p>It validates storage properties as follows
+   *
+   * <p>1. default-type for storage is not set and storage types are not provided, ie. LocalStorage
+   * is configured **valid**
+   *
+   * <p>2. default-type for storage is set and storage types are provided ,and it contains
+   * default-type **valid**
+   *
+   * <p>all other configurations are **invalid**
+   */
+  @PostConstruct
+  public void validateProperties() {
+    String clusterYamlError = "Cluster yaml is incorrectly configured: ";
+    Preconditions.checkArgument(
+        !(storageProperties.getDefaultType() == null
```

Comment:
Would this be better:

if (storageProperties.getDefaultType() == null || storageProperties.getDefaultType().isEmpty()) {
  // Default type not configured. Storage types should also be null or empty.
  Preconditions.checkArgument(storageProperties.getTypes() == null || storageProperties.getTypes().isEmpty(), "");
} else {
  // The default type must be one of the configured storage types.
  Preconditions.checkArgument(storageProperties.getTypes() != null && !storageProperties.getTypes.isEmpty() && storageProperties.getTypes().containsKey(storageProperties.getDefaultType()), "");
}

Reply:
yes better! thank you.